market-generator/src/test/java/market/MarketSubscriptionManagerTest.java

```diff
@@ -0,0 +1,159 @@
+package market;
+
+import com.market.MarketDataRequest;
+import com.market.TickResponse;
+import com.market.banica.generator.exception.NoSuchGoodException;
+import com.market.banica.generator.service.MarketSubscriptionManager;
+import io.grpc.stub.StreamObserver;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.HashSet;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+class MarketSubscriptionManagerTest {
+
+    @InjectMocks
+    private MarketSubscriptionManager marketSubscriptionManager;
+
+    private final StreamObserver<TickResponse> subscriberOne = mock(StreamObserver.class);
+
+    private final StreamObserver<TickResponse> subscriberTwo = mock(StreamObserver.class);
+
+    private static final String GOOD_BANICA = "Banica";
+
+    private static final String GOOD_EGGS = "Eggs";
+
+    private static final MarketDataRequest MARKET_DATA_REQUEST_BANICA = MarketDataRequest.newBuilder().setGoodName(GOOD_BANICA).build();
+
+    private static final MarketDataRequest MARKET_DATA_REQUEST_INVALID = MarketDataRequest.newBuilder().build();
+
+
+    @Test
+    void subscribeShouldAddNewSubscriberToSubscriptions() {
```

Comment:
Suggestion: Use the following convention for test names: "whatTheTestDoes/WhatParamsItTakes/Result
Example: subscribeForItemWithValidDataInputExcutesWithSuccess

This comment is valid for all test names.